src/mutex.rs

```diff
@@ -123,6 +123,30 @@ impl<T> BusMutex for CortexMMutex<T> {
     }
 }
 
+/// Alias for xtensa-lx6 mutex.
+///
+/// Based on [`xtensa-lx6::mutex::CriticalSectionSpinLockMutex`][xtensa-lx6-mutex].  This mutex works by disabling
+/// interrupts while the mutex is locked.
+///
+/// [xtensa-lx6-mutex]: https://docs.rs/xtensa-lx6/0.2.0/xtensa_lx6/mutex/struct.CriticalSectionSpinLockMutex.html
+///
+/// This type is only available with the `xtensa_lx6` feature.
+#[cfg(feature = "xtensa-lx6")]
+pub type XtensaMutex<T> = spin::Mutex<T>;
```

Comment:
In the documentation you're linking a Mutex implementation from the `xtensa-lx6` crate, but here you're referencing once from `spin`.  What's the reason behind that?  Should we be using the `CriticalSectionSpinLockMutex` instead?

Reply:
It's some left over from a previous try, I'll fix the documentation.

Underneath, `CriticalSectionSpinLockMutex` from `xtensa-lx6` uses `spin::Mutex` but the `lock` API exposed by `CriticalSectionSpinLockMutex` requires a [mutable reference](https://docs.rs/xtensa-lx6/0.2.0/xtensa_lx6/mutex/trait.Mutex.html#tymethod.lock).

Reply:
If I see this correctly, the `Mutex` trait is implemented for `&CriticalSectionSpinLockMutex` which means that you actually _can_ get a mutable `self` here (= `&mut &CriticalSectionSpinLockMutex`).  But this is all quite convoluted indeed.

The only problem I see with your approach is that other platforms which also want to use `spin::Mutex` can't, because you've already type-aliased it for xtensa... Maybe a newtype is a better option here?

Reply:
> If I see this correctly, the Mutex trait is implemented for &CriticalSectionSpinLockMutex which means that you actually can get a mutable self here (= &mut &CriticalSectionSpinLockMutex). But this is all quite convoluted indeed.

I took some time to try this but did not succeed, I'm still quite new to Rust and must admit that these kind of things is still a bit difficult for me.

> The only problem I see with your approach is that other platforms which also want to use spin::Mutex can't, because you've already type-aliased it for xtensa... Maybe a newtype is a better option here?

I thought that the type aliasing would be hidden behind the feature gate. Am I wrong ?

Reply:
> I took some time to try this but did not succeed, I'm still quite new to Rust and must admit that these kind of things is still a bit difficult for me.

Don't worry, a mutable reference to an immutable reference is a very unintuitive thing to deal with...

> I thought that the type aliasing would be hidden behind the feature gate. Am I wrong ?

At its core, this is true - but feature-flags should be designed to be additive only, not exclusive.  Meaning that code should also compile with `--all-features` if at all possible.  This wouldn't work with the alias.

Reply:
Thanks for your answers and for your time.

So if I understand correctly, I should do something like this:
```rust
pub struct XtensaMutex<T>(spin::Mutex<T>);
```

Reply:
Exactly!